packages/x-date-pickers/src/internals/hooks/useField/useField.ts

```diff
@@ -0,0 +1,389 @@
+  setSectionValue,
+} from './useField.utils';
+
+export const useField = <
+  TInputValue,
+  TValue,
+  TDate,
+  TSection extends FieldSection,
+  TProps extends UseFieldProps<TInputValue, TValue, InferError<TProps>>,
+>(
+  params: UseFieldParams<TInputValue, TValue, TDate, TSection, TProps>,
+): UseFieldResponse<TProps> => {
+  const utils = useUtils<TDate>();
+  const inputRef = React.useRef<HTMLInputElement>(null);
+
+  const {
+    props: {
+      value: valueProp,
+      defaultValue,
+      onChange,
+      onError,
+      format = utils.formats.keyboardDate,
+      readOnly = false,
+      ...otherProps
+    },
+    valueManager,
+    fieldValueManager,
+    validator,
+  } = params;
+
+  const firstDefaultValue = React.useRef(defaultValue);
+
+  const valueParsed = React.useMemo(() => {
+    // TODO: Avoid this type casting, the emptyValues are both valid TDate and TInputDate
+    const value =
+      firstDefaultValue.current ?? valueProp ?? (valueManager.emptyValue as unknown as TInputValue);
+
+    return valueManager.parseInput(utils, value);
+  }, [valueProp, valueManager, utils]);
+
+  const [state, setState] = React.useState<UseFieldState<TValue, TSection[]>>(() => {
+    const sections = fieldValueManager.getSectionsFromValue(utils, null, valueParsed, format);
+
+    return {
+      sections,
+      valueParsed,
+      valueStr: fieldValueManager.getValueStrFromSections(sections),
+      selectedSectionIndexes: null,
+    };
+  });
+
+  const updateSections = (sections: TSection[]) => {
+    const { value: newValueParsed, shouldPublish } = fieldValueManager.getValueFromSections(
+      utils,
+      state.sections,
+      sections,
+      format,
+    );
+
+    setState((prevState) => ({
+      ...prevState,
+      sections,
+      valueStr: fieldValueManager.getValueStrFromSections(sections),
+      valueParsed: newValueParsed,
+    }));
+
+    if (onChange && shouldPublish) {
+      onChange(newValueParsed);
+    }
+  };
+
+  const updateSelectedSections = (start?: number, end?: number) => {
+    setState((prevState) => ({
+      ...prevState,
+      selectedSectionIndexes: start == null ? null : { start, end: end ?? start },
+      selectedSectionQuery: null,
+    }));
+  };
+
+  const handleInputClick = useEventCallback(() => {
+    if (state.sections.length === 0) {
+      return;
+    }
+
+    const nextSectionIndex = state.sections.findIndex(
+      (section) => section.start > (inputRef.current?.selectionStart ?? 0),
+    );
+    const sectionIndex = nextSectionIndex === -1 ? state.sections.length - 1 : nextSectionIndex - 1;
+
+    updateSelectedSections(sectionIndex);
+  });
+
+  const handleInputKeyDown = useEventCallback((event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (!inputRef.current || state.sections.length === 0) {
+      return;
+    }
+
+    // eslint-disable-next-line default-case
+    switch (true) {
+      // Select all
+      case event.key === 'a' && event.ctrlKey: {
```

Comment:
In order to support both `Ctrl + A` and Mac specific `Command + A`:
```suggestion
      case event.key === 'a' && (event.ctrlKey || event.metaKey): {
```